sambacc/commands/ctdb.py

```diff
@@ -261,12 +261,48 @@ def ctdb_migrate(ctx: Context) -> None:
         ctdb.archive_tdb(ctx.instance_config, ctx.cli.archive)
 
 
-def _lookup_hostname(hostname):
-    # XXX this is a nasty little hack.
-    ips = socket.gethostbyname_ex(hostname)[2]
-    addr = [ip for ip in ips if ip != "127.0.0.1"][0]
-    _logger.info(f"Determined address for {hostname}: {addr}")
-    return addr
+def _lookup_hostname(hostname: str) -> str:
+    try:
+        addrinfo = socket.getaddrinfo(
+            hostname, None, family=socket.AF_UNSPEC, type=socket.SOCK_STREAM
+            )
+        ipv4_addresses = []
+        ipv6_addresses = []
+
+        for entry in addrinfo:
+            family, _, _, _, sockaddr = entry
+            ip_address = sockaddr[0]
+
+            if ip_address.startswith("127.") or ip_address == "::1":
+                continue
+
+            if family == socket.AF_INET:
+                ipv4_addresses.append(ip_address)
```

Comment:
We are going to return the first IPv4 address if it exists, so there's no need to continue checking other addresses. We can directly return the current address from here.

Reply:
I see! I think once I remove keeeping track of list it'll get fixed since I'll directly return first address whether it's ipv4 or piv6

Reply:
I see! I think once I remove keeeping track of list it'll get fixed since I'll directly return first address whether it's ipv4 or ipv6